# ECON315: INTERNATIONAL TRADE - PROBLEM SET 1


Here is the [raw dataset](https://drive.google.com/drive/folders/1B5hddr0M5E9ko9G3VlIT3mKoqAoP0m-g?usp=sharing) I used. But I will process to have all data of Southeast Asia countries in 2019.

## Import the dataset

In [ ]:
import pandas as pd
import matplotlib as plt

In [ ]:
export1 = pd.read_csv('/content/export_file1.csv')
display(export1.head())

reporter_name reporter_code  year classification classification_version  \
0      Viet Nam          C704  2019            MTN                    MT4   
1      Viet Nam          C704  2019            MTN                    MT4   
2      Thailand          C764  2019            MTN                    MT4   
3       Myanmar          C104  2019            MTN                    MT4   
4      Cambodia          C116  2019            MTN                    MT4   

  product_code mtn_categories partner_code                       partner_name  \
0        TOTAL   All products         C792                            Türkiye   
1        TOTAL   All products         C024                             Angola   
2        TOTAL   All products         C558                          Nicaragua   
3        TOTAL   All products         C862  Venezuela, Bolivarian Republic of   
4        TOTAL   All products         C012                            Algeria   

          value  
0  1.140412e+09  
1  4.268524e+07  
2  1.228452e+07  
3  2.489294e+06  
4  1.783262e+04

In [ ]:
export2 = pd.read_csv('/content/export_file2.csv')
display(export2.head())

reporter_name reporter_code  year classification classification_version  \
0     Indonesia          C360  2019            MTN                    MT4   
1   Philippines          C608  2019            MTN                    MT4   
2      Malaysia          C458  2019            MTN                    MT4   
3      Malaysia          C458  2019            MTN                    MT4   
4   Philippines          C608  2019            MTN                    MT4   

  product_code mtn_categories partner_code              partner_name  \
0        TOTAL   All products         C682  Saudi Arabia, Kingdom of   
1        TOTAL   All products         C324                    Guinea   
2        TOTAL   All products         C484                    Mexico   
3        TOTAL   All products         C704                  Viet Nam   
4        TOTAL   All products         C586                  Pakistan   

          value  
0  1.503159e+09  
1  9.225570e+05  
2  2.185864e+09  
3  8.542448e+09  
4  4.913324e+07

In [ ]:
export3 = pd.read_csv('/content/export_file3.csv')
display(export3.head())

reporter_name reporter_code  year classification classification_version  \
0   Timor-Leste          C626  2019            MTN                    MT4   
1   Timor-Leste          C626  2019            MTN                    MT4   
2   Timor-Leste          C626  2019            MTN                    MT4   
3   Timor-Leste          C626  2019            MTN                    MT4   
4   Timor-Leste          C626  2019            MTN                    MT4   

  product_code mtn_categories partner_code        partner_name       value  
0        TOTAL   All products         C410  Korea, Republic of   307537.15  
1        TOTAL   All products         U918      European Union  2077280.20  
2        TOTAL   All products         C356               India       26.25  
3        TOTAL   All products         C158      Chinese Taipei   320437.32  
4        TOTAL   All products         C360           Indonesia  6730025.70

In [ ]:
import1 = pd.read_csv('/content/import_file1.csv')
display(import1.head())

reporter_name reporter_code  year classification classification_version  \
0      Viet Nam          C704  2019            MTN                    MT4   
1      Viet Nam          C704  2019            MTN                    MT4   
2      Cambodia          C116  2019            MTN                    MT4   
3      Cambodia          C116  2019            MTN                    MT4   
4       Myanmar          C104  2019            MTN                    MT4   

  product_code mtn_categories partner_code              partner_name  \
0        TOTAL   All products         C156                     China   
1        TOTAL   All products         C840  United States of America   
2        TOTAL   All products         C454                    Malawi   
3        TOTAL   All products         C854              Burkina Faso   
4        TOTAL   All products         C524                     Nepal   

          value  
0  7.558610e+10  
1  1.437681e+10  
2  9.812880e+05  
3  3.119760e+04  
4  1.327180e+03

In [ ]:
import2 = pd.read_csv('/content/import_file2.csv')
display(import2.head())

reporter_name reporter_code  year classification  \
0  Brunei Darussalam          C096  2019            MTN   
1          Indonesia          C360  2019            MTN   
2          Indonesia          C360  2019            MTN   
3           Malaysia          C458  2019            MTN   
4          Indonesia          C360  2019            MTN   

  classification_version product_code mtn_categories partner_code  \
0                    MT4        TOTAL   All products         C332   
1                    MT4        TOTAL   All products         C048   
2                    MT4        TOTAL   All products         C430   
3                    MT4        TOTAL   All products         C826   
4                    MT4        TOTAL   All products         C212   

          partner_name         value  
0                Haiti  4.999199e+02  
1  Bahrain, Kingdom of  1.789420e+08  
2              Liberia  1.680812e+07  
3       United Kingdom  1.748966e+09  
4             Dominica  1.633300e+04

In [ ]:
import3 = pd.read_csv('/content/import_file3.csv')
display(import3.head())

reporter_name reporter_code  year classification classification_version  \
0   Timor-Leste          C626  2019            MTN                    MT4   
1   Timor-Leste          C626  2019            MTN                    MT4   
2   Timor-Leste          C626  2019            MTN                    MT4   
3   Timor-Leste          C626  2019            MTN                    MT4   
4   Timor-Leste          C626  2019            MTN                    MT4   

  product_code mtn_categories partner_code partner_name      value  
0        TOTAL   All products         C554  New Zealand  960699.65  
1        TOTAL   All products         C268      Georgia     326.91  
2        TOTAL   All products         C608  Philippines  519542.34  
3        TOTAL   All products         C104      Myanmar  510138.19  
4        TOTAL   All products         C894       Zambia    7308.00

In [ ]:
gdp = pd.read_csv('/content/GDP_WorldBank.csv', skiprows=4)
#gdp = gdp.dropna()
display(gdp.head())

Country Name Country Code     Indicator Name  \
0                        Aruba          ABW  GDP (current US$)   
1  Africa Eastern and Southern          AFE  GDP (current US$)   
2                  Afghanistan          AFG  GDP (current US$)   
3   Africa Western and Central          AFW  GDP (current US$)   
4                       Angola          AGO  GDP (current US$)   

   Indicator Code          1960          1961          1962          1963  \
0  NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   
1  NY.GDP.MKTP.CD  2.420993e+10  2.496326e+10  2.707802e+10  3.177483e+10   
2  NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   
3  NY.GDP.MKTP.CD  1.190511e+10  1.270803e+10  1.363092e+10  1.446926e+10   
4  NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   

           1964          1965  ...          2016          2017          2018  \
0           NaN           NaN  ...  2.983635e+09  3.092429e+09  3.276184e+09   
1  3.028492e+10  3.381219e+10  ...  8.289612e+11  9.730251e+11  1.012291e+12   
2           NaN           NaN  ...  1.811657e+10  1.875346e+10  1.805322e+10   
3  1.580394e+10  1.692124e+10  ...  7.000282e+11  6.940513e+11  7.778404e+11   
4           NaN           NaN  ...  5.276162e+10  7.369015e+10  7.945069e+10   

           2019          2020          2021          2022          2023  \
0  3.395799e+09  2.481857e+09  2.929447e+09  3.279344e+09  3.648573e+09   
1  1.009747e+12  9.334072e+11  1.085605e+12  1.191639e+12  1.176910e+12   
2  1.879944e+10  1.995593e+10  1.426000e+10  1.449724e+10  1.715223e+10   
3  8.332889e+11  7.972952e+11  8.581145e+11  8.936399e+11  8.147285e+11   
4  7.089796e+10  4.850156e+10  6.650513e+10  1.043997e+11  8.487516e+10   

           2024  Unnamed: 69  
0           NaN          NaN  
1  1.287677e+12          NaN  
2           NaN          NaN  
3  6.700257e+11          NaN  
4  8.039694e+10          NaN  

[5 rows x 70 columns]

## Add up each country's GDP to obtain Southeast Asia's GDP

In [ ]:
sea_country = ['Lao PDR', 'Thailand', 'Cambodia', 'Brunei Darussalam', 'Malaysia', 'Singapore', 'Viet Nam', 'Philippines', 'Myanmar', 'Indonesia', 'Timor-Leste']
sea_country_gdp = gdp[gdp['Country Name'].isin(sea_country)][['Country Name', '2019']]
display(sea_country_gdp)

Country Name          2019
31   Brunei Darussalam  1.346924e+10
106          Indonesia  1.119100e+12
123           Cambodia  3.668536e+10
129            Lao PDR  1.874056e+10
160            Myanmar  7.506511e+10
169           Malaysia  3.651777e+11
187        Philippines  3.768234e+11
208          Singapore  3.761620e+11
233           Thailand  5.439767e+11
237        Timor-Leste  2.032550e+09
257           Viet Nam  3.343653e+11

In [ ]:
sea_asia_gdp_2019 = sea_country_gdp['2019'].sum()
print(f"Southeast Asia's GDP in 2019: {sea_asia_gdp_2019}")

Southeast Asia's GDP in 2019: 3261597765617.0527


SEA's GDP in 2019 (current $US\$$) is 3,261,597,765,617.0527 USD.
***(estimated 3.261 thousand billion USD)***.

## Divide each country's GDP by Southeast Asia's GDP to obtain its share of Southeast Asia's GDP.

In [ ]:
sea_country_gdp['Percentage Share'] = (sea_country_gdp['2019'] / sea_asia_gdp_2019) * 100
display(sea_country_gdp)

Country Name          2019  Percentage Share
31   Brunei Darussalam  1.346924e+10          0.412964
106          Indonesia  1.119100e+12         34.311400
123           Cambodia  3.668536e+10          1.124766
129            Lao PDR  1.874056e+10          0.574582
160            Myanmar  7.506511e+10          2.301483
169           Malaysia  3.651777e+11         11.196283
187        Philippines  3.768234e+11         11.553338
208          Singapore  3.761620e+11         11.533059
233           Thailand  5.439767e+11         16.678227
237        Timor-Leste  2.032550e+09          0.062318
257           Viet Nam  3.343653e+11         10.251579

We can figure out that by GDP, Southeast Asia is divided into two tiers:
- Tier 1: More developing countries with higher GDP share (Indonesia, Thailand, Philippines, Singapore, Malaysia, Vietnam)
- Tier 2: Less developing countries (the remaining, contributing under $3\%$ of the GDP).

However, the metric is relative as we have not yet calculated GDP per capita.

## For each country, add the value of its exports with Vietnam and the value of its imports from Vietnam to get its trade volume with Vietnam.

In [ ]:
# Exporters from file export_1
vietnam_exports1 = export1[export1['partner_name'] == 'Viet Nam'][['reporter_name', 'partner_name', 'value']]
display(vietnam_exports1)

reporter_name partner_name         value
133                          Thailand     Viet Nam  1.211492e+10
144  Lao People's Democratic Republic     Viet Nam  1.054900e+09
259                          Cambodia     Viet Nam  3.591957e+08
412                           Myanmar     Viet Nam  2.300927e+08

In [ ]:
# Exporters from file export_2
vietnam_exports2 = export2[export2['partner_name'] == 'Viet Nam'][['reporter_name', 'partner_name', 'value']]
display(vietnam_exports2)

reporter_name partner_name         value
3             Malaysia     Viet Nam  8.542448e+09
303          Singapore     Viet Nam  1.296125e+10
386          Indonesia     Viet Nam  5.153358e+09
542  Brunei Darussalam     Viet Nam  1.987429e+08
635        Philippines     Viet Nam  1.269636e+09

In [ ]:
# Exporters from file export_3
vietnam_exports3 = export3[export3['partner_name'] == 'Viet Nam'][['reporter_name', 'partner_name', 'value']]
display(vietnam_exports3)

reporter_name partner_name     value
11   Timor-Leste     Viet Nam  13652.32

In [ ]:
# Exporters from file import_1
vietnam_imports1 = import1[import1['partner_name'] == 'Viet Nam'][['reporter_name', 'partner_name', 'value']]
display(vietnam_imports1)

reporter_name partner_name         value
116  Lao People's Democratic Republic     Viet Nam  4.457159e+08
151                          Thailand     Viet Nam  5.443781e+09
217                           Myanmar     Viet Nam  6.249083e+08
234                          Cambodia     Viet Nam  2.718188e+09

In [ ]:
# Exporters from file import_2
vietnam_imports2 = import2[import2['partner_name'] == 'Viet Nam'][['reporter_name', 'partner_name', 'value']]
display(vietnam_imports2)

reporter_name partner_name         value
89            Malaysia     Viet Nam  4.660183e+09
180          Singapore     Viet Nam  3.646541e+09
531          Indonesia     Viet Nam  3.848196e+09
628  Brunei Darussalam     Viet Nam  1.153181e+08
729        Philippines     Viet Nam  3.860033e+09

In [ ]:
# Exporters from file import_3
vietnam_imports3 = import3[import3['partner_name'] == 'Viet Nam'][['reporter_name', 'partner_name', 'value']]
display(vietnam_imports3)

reporter_name partner_name        value
33   Timor-Leste     Viet Nam  28610683.56

In [ ]:
# Combine and sum export values
combined_vietnam_exports = pd.concat([vietnam_exports1, vietnam_exports2, vietnam_exports3])
vietnam_export_summary = combined_vietnam_exports.groupby('reporter_name')['value'].sum().reset_index()
vietnam_export_summary = vietnam_export_summary.rename(columns={'value': 'export_value (VN)'})

# Combine and sum import values
combined_vietnam_imports = pd.concat([vietnam_imports1, vietnam_imports2, vietnam_imports3])
vietnam_import_summary = combined_vietnam_imports.groupby('reporter_name')['value'].sum().reset_index()
vietnam_import_summary = vietnam_import_summary.rename(columns={'value': 'import_value (VN)'})

# Merge export and import summaries
vietnam_trade_summary = pd.merge(vietnam_export_summary, vietnam_import_summary, on='reporter_name', how='outer').fillna(0)

# Calculate trade volume
vietnam_trade_summary['trade_volume (VN)'] = vietnam_trade_summary['export_value (VN)'] + vietnam_trade_summary['import_value (VN)']

# Display the desired columns
display(vietnam_trade_summary[['reporter_name', 'export_value (VN)', 'import_value (VN)', 'trade_volume (VN)']])

reporter_name  export_value (VN)  import_value (VN)  \
0                 Brunei Darussalam       1.987429e+08       1.153181e+08   
1                          Cambodia       3.591957e+08       2.718188e+09   
2                         Indonesia       5.153358e+09       3.848196e+09   
3  Lao People's Democratic Republic       1.054900e+09       4.457159e+08   
4                          Malaysia       8.542448e+09       4.660183e+09   
5                           Myanmar       2.300927e+08       6.249083e+08   
6                       Philippines       1.269636e+09       3.860033e+09   
7                         Singapore       1.296125e+10       3.646541e+09   
8                          Thailand       1.211492e+10       5.443781e+09   
9                       Timor-Leste       1.365232e+04       2.861068e+07   

   trade_volume (VN)  
0       3.140610e+08  
1       3.077383e+09  
2       9.001554e+09  
3       1.500616e+09  
4       1.320263e+10  
5       8.550010e+08  
6       5.129669e+09  
7       1.660779e+10  
8       1.755870e+10  
9       2.862434e+07

## Add up each country's trade volume with Vietnam to get Southeast Asia's trade volume with Vietnam.

In [ ]:
vietnam_trade_summary_2019 = vietnam_trade_summary['trade_volume (VN)'].sum()
print(f"Southeast Asia's trade volume with Vietnam is: {vietnam_trade_summary_2019}")

Southeast Asia's trade volume with Vietnam is: 67276029575.22916


## Divide each country's trade volume with Vietnam by Southeast Asia's trade volume with Vietnam to obtain its share of Southeast Asia's trade with Vietnam.

In [ ]:
vietnam_trade_summary['Percentage Share'] = (vietnam_trade_summary['trade_volume (VN)']/vietnam_trade_summary_2019) * 100
display(vietnam_trade_summary)

reporter_name  export_value (VN)  import_value (VN)  \
0                 Brunei Darussalam       1.987429e+08       1.153181e+08   
1                          Cambodia       3.591957e+08       2.718188e+09   
2                         Indonesia       5.153358e+09       3.848196e+09   
3  Lao People's Democratic Republic       1.054900e+09       4.457159e+08   
4                          Malaysia       8.542448e+09       4.660183e+09   
5                           Myanmar       2.300927e+08       6.249083e+08   
6                       Philippines       1.269636e+09       3.860033e+09   
7                         Singapore       1.296125e+10       3.646541e+09   
8                          Thailand       1.211492e+10       5.443781e+09   
9                       Timor-Leste       1.365232e+04       2.861068e+07   

   trade_volume (VN)  Percentage Share  
0       3.140610e+08          0.466825  
1       3.077383e+09          4.574264  
2       9.001554e+09         13.380032  
3       1.500616e+09          2.230535  
4       1.320263e+10         19.624570  
5       8.550010e+08          1.270885  
6       5.129669e+09          7.624809  
7       1.660779e+10         24.686038  
8       1.755870e+10         26.099495  
9       2.862434e+07          0.042548

In [ ]:
import altair as alt

# Sort the data by percentage share in descending order (still useful for order)
vietnam_trade_summary_sorted = vietnam_trade_summary.sort_values(by='Percentage Share', ascending=False)

# Create the base chart
base = alt.Chart(vietnam_trade_summary_sorted).encode(
    theta=alt.Theta("trade_volume (VN)", stack=True) # Removed sort="-theta" in previous fix
)

# Create the pie chart - Adding a subtle stroke for a slight 3D effect
pie = base.mark_arc(outerRadius=120, stroke="#ffffff", strokeWidth=1).encode(
    color=alt.Color(
        "Percentage Share",
        type="quantitative", # Encode a quantitative field
        scale=alt.Scale(range="heatmap"), # Use a sequential scheme where darker means higher values
        legend=alt.Legend(title="Trade Share (%)") # Add a legend for the color scale
    ),
    order=alt.Order("trade_volume (VN)", sort="descending"), # Order slices by size
    tooltip=[
        "reporter_name",
        alt.Tooltip("trade_volume (VN)", title="Trade Volume (USD)"),
        alt.Tooltip("Percentage Share", format=".2f", title="Percentage Share (%)")
    ]
)

# Add text labels (Country Name) for top 4 only
top_4_countries = vietnam_trade_summary_sorted.head(4)['reporter_name'].tolist()
text = base.mark_text(radius=140).encode(
    text=alt.Text("reporter_name"), # Display country name
    order=alt.Order("trade_volume (VN)", sort="descending"),
    color=alt.value("black"), # Text color
    opacity=alt.condition(
        alt.FieldOneOfPredicate(field='reporter_name', oneOf=top_4_countries),
        alt.value(1), # Show text for top 4
        alt.value(0)  # Hide text for others
    )
)


# Combine the pie chart and text labels and add title and font
chart = (pie + text).properties(
    title="Percentage Share of Trade with Vietnam (2019)"
).configure_text(
    font='lato'
)

# Display the chart
chart.display()

alt.LayerChart(...)